pkg/odo/cli/preference/registry/util/util.go

```diff
@@ -28,10 +30,6 @@ func IsSecure(prefClient preference.Client, registryName string) bool {
 	return isSecure
 }
 
-func IsGitBasedRegistry(url string) bool {
+func IsGithubBasedRegistry(url string) bool {
 	return strings.Contains(url, "github.com") || strings.Contains(url, "raw.githubusercontent.com")
```

Comment:
Shouldn't we parse the URL string and test the host part instead? 
I find it weird to check that the URL just contains those strings. Because there is nothing preventing someone from using a URL like `https://my.registry.example.com/github.com`. This would get rejected by this function, no?

Reply:
hmm valid point. I never thought of it that way